File: lessons/lesson02/src/my_publisher.cpp
#include <chrono>
#include <memory>
#include <string>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"
#include "sensor_msgs/msg/image.hpp"

using namespace std::chrono_literals;

class MyPublisher : public rclcpp::Node
{
public:
  MyPublisher() : Node("my_publisher")
  {
    // TODO: Create your publishers here

    // TODO: Create your timers here (adjust periods for desired frequencies)

    
    RCLCPP_INFO(this->get_logger(), "Publisher node started");
  }

private:
  void timer_callback1()
  {
    // TODO: Create and publish Image message to topic_alpha

  }
  
  void timer_callback2()
  {
    // TODO: Create and publish String message to topic_beta

  }
  
  rclcpp::TimerBase::SharedPtr timer1_;
  rclcpp::TimerBase::SharedPtr timer2_;
  rclcpp::Publisher<sensor_msgs::msg::Image>::SharedPtr publisher_alpha_;
  rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_beta_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<MyPublisher>());
  rclcpp::shutdown();
  return 0;
}

File: lessons/lesson02/lesson02/my_publisher.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from std_msgs.msg import String
from sensor_msgs.msg import Image


class MyPublisher(Node):
    def __init__(self):
        super().__init__('my_publisher')
        
        # TODO: Create your publishers here

        # TODO: Create your timers here (adjust periods for desired frequencies)

        
        self.get_logger().info('Publisher node started')
    
    def timer_callback1(self):
        # TODO: Create and publish Image message to topic_alpha

        pass
    
    def timer_callback2(self):
        # TODO: Create and publish String message to topic_beta

        pass


def main(args=None):
    rclpy.init(args=args)
    node = MyPublisher()
    
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        rclpy.shutdown()


if __name__ == '__main__':
    main()
